code/create/dta_dicom2spec_rules.py

```diff
@@ -0,0 +1,226 @@
+    if protocol:
+
+        # BEGIN Additional rule for forrest-structural
+        # TODO: Probably to be moved to some rule enhancement
+        if "VEN_BOLD" in protocol:
+            # TODO: Not clear yet; swi might be considered a datatype rather than
+            # a modality by respective BIDS extension:
+            # https://docs.google.com/document/d/1kyw9mGgacNqeMbp4xZet3RnDhcMmf4_BmRgKaOkO2Sc
+            return "swi"
+
+        if "DTI_" in protocol:
+            # TODO: What actually is the relevant part of protocol here?
+            return "dwi"
+
+        if "60dir".lower() in protocol.lower():
+            return "dwi"
+        if "120dir".lower() in protocol.lower():
+            return "dwi"
+
+        if "field map" in protocol:
+            return "fieldmap"
+        # END
+
+        import re
+        prot_parts = re.split('_|-|\s', protocol.lower())
+
+        # this is a bad protocol name, but it is a resting state scan for pmu_rest and a task-bold if bold_task
+        if protocol.lower() in "dta_epi_pmu_rest" or protocol.lower() in "DTA_ep2d_bold_Task".lower():
+               return "bold"
+
+########## Hier fuer DTA extra die Abfrage machen, ob Magnitude or phasediff
+        image_type = record.get("ImageType", None)
+        if image_type and "fieldmap" in protocol.lower():
+                return "fieldmap"
+
+
+        # I deleted "t2star" from that list for this specific dataset, because it does not contain these.
+        # It would probably also be fine, if I had moved the "fieldmap" value to the start
+        direct_search_terms = ["t1", "t1w", "t2", "t2w",
+                               "t1rho", "t1map", "t2map", "flair",
+                               "flash", "pd", "pdmap", "pdt2", "inplanet1",
+                               "inplanet2", "angio", "dwi", "phasediff",
+                               "phase1", "phase2", "magnitude1", "magnitude2",
+                               "fieldmap", "epi", "meg"]
+
+        for m in direct_search_terms:
+            if m in prot_parts:
+                return m
+            if "mprage" in prot_parts:
+                return "t1w"
+
+        # BEGIN: Additional rule for forrest-structural
+        # TODO: Probably to be moved to some rule enhancement
+        if "st1w" in prot_parts:
+            return "t1w"
+        if "st2w" in prot_parts:
+            return "t2w"
+        if "tof" in prot_parts:
+            return "angio"
+        # END
+
+        if "func" in prot_parts:
+            return "bold"
+
+    # found nothing, but modality isn't necessarily required
+    return None
+
+def _guess_task(dcm_dict, record):
+    protocol = record.get("ProtocolName", None)
+    count_if_same_protocol_but_scanned_earlier = 1
+    #
+    exp_flag = False
+
+    if protocol:
+        if protocol.lower() in "dta_epi_pmu_rest":
+               return "rest"
+        if protocol.lower() in "DTA_ep2d_bold_Task".lower():
+            for elem in dcm_dict._dicom_series:
```

Comment:
Why do you loop over other image series here? I'm not sure I understand what your goal is. Why not use `record`?